test/napi_throw/test_napi_throw.c

```diff
@@ -0,0 +1,28 @@
+#include <assert.h>
+
+#include <node_api.h>
+
+napi_value ThrowError(napi_env env, napi_callback_info info) {
+  napi_throw_error(env, NULL, "an error occurred"); 
+  return NULL;
+}
+
+static napi_value Init(napi_env env, napi_value exports) {
+  napi_value result;
+  assert(napi_create_object(env, &result) == napi_ok);
```

Comment:
My understanding is that under certain build circumstances arguments to `assert()` are ignored. So, in that case, `napi_create_object()` won't be called at all. It might be better to
```C
napi_status status = napi_create_object(env, &result);
assert(status == napi_ok);
```

Reply:
I'll change. I based this on the example `addon_node.c` from the docs: https://nodejs.org/dist/latest-v12.x/docs/api/n-api.html#n_api_n_api_version_matrix

Reply:
Now using separate `status` variable and `assert()` checks.